packages/angular-cli/utilities/check-package-manager.ts

```diff
@@ -0,0 +1,60 @@
+import * as chalk from 'chalk';
+import {exec} from 'child_process';
+import {CliConfig} from '../models/config';
+
+const Promise = require('../ember-cli/lib/ext/promise');
+const config = CliConfig.fromProject();
+
+let packageManager = 'npm';
+if (config && config.get('packageManager')) {
+  packageManager = config.get('packageManager');
+}
+
+export function checkYarnOrCNPM() {
+  return new Promise((resolve, reject) => {
+    Promise
+      .all([checkYarn(), checkCNPM()])
+      .then((data) => {
+        if (packageManager === 'npm') {
+          const isYarnInstalled = data[0];
+          const isCNPMInstalled = data[1];
+          if (isYarnInstalled && isCNPMInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=yarn` ' +
+            'or `ng set --global packageManager=cnpm`'));
+          } else if (isYarnInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=yarn`'));
+          } else if (isCNPMInstalled) {
+            console.log(chalk.yellow('you can `ng set --global packageManager=cnpm`'));
+          }
+        }
+        resolve();
+      })
+      .catch(resolve);
+  });
+}
+
+function checkYarn() {
+  return new Promise(function(resolve, reject) {
```

Comment:
You can use `Promise.denodeify` for this (look up elsewhere in the code where we use it);

```typescript
const exec = Promise.denodeify(fs.exec);

function checkYarn() {
  return exec('yarn --version')
    .then(() => true, () => false);
}
```

Much cleaner :)